core/ui/EditorToolbar/picture-dropdown.tid

```diff
@@ -4,8 +4,44 @@ title: $:/core/ui/EditorToolbar/picture-dropdown
 [img[$(imageTitle)$]]
 \end
 
+\define insert-external-image()
```

Comment:
Limiting the displayed thumbnails number seems like a good compromise.
Not only before typing (maybe sorted by creation date, most recent displayed only) but also for search result matches after the user types something (sorted by relevance).

Perhaps even a lower number around the twenties is advisable for lower end devices. 

Thanks for the feedback, and taking interest in this.